app/admin/books.rb

```diff
@@ -12,23 +25,48 @@
     end
   end
 
-  show do |book|
+  member_action :history do
+    @versioned_object = Book.find(params[:id])
+    @versions = PaperTrail::Version.where(item_type: 'Book', item_id: @versioned_object.id).order('created_at ASC')
+    render partial: 'admin/history'
+  end
 
+  show title: Proc.new{
+      book_with_versions = Book.includes(versions: :item).find(params[:id])
+      if params[:version]
+        # title of a version of the book
+        version = book_with_versions.versions[(params[:version].to_i - 1).to_i].reify
+        version.title
+      else
+        # the current book title
+        book_with_versions.title
+      end
+    } do |book|
     attributes_table do
       [:title, :sub_title, :format, :publication_date, :isbn, :primary_language, :call_number, :description, :physical_description, :notes, :statement_of_responsibility, :created_at, :updated_at].each do |prop|
         row prop
       end
     end
 
+    return if params[:version] == '0'
+
+    # choose which version's data to display
+    if params[:version]
+      book_with_versions = Book.includes(versions: :item).find(params[:id])
+      book_version = book_with_versions.versions[(params[:version].to_i - 1).to_i].reify
```

Comment:
It takes the JSON stored in the `object` field of the version and builds a temporary object out of it so that you can reference the attributes normally such as with `book.title`.  Otherwise it's something like this: `JSON.parse(version.object)[:title]`.
```
=> #<PaperTrail::Version id: 122, item_type: "TeacherSet", item_id: 20830710052907, event: "update", whodunnit: nil, object: "---\nid: 20830710052907\ntitle: Animal Compare and Co...", created_at: "2018-09-26 14:24:03", object_changes: "---\nlast_book_change:\n- \n- 'updated-20599992052-Vam...">
```

Reply:
Is there a risk of exceptions being thrown?  (does this need to be in a try-catch?)

Reply:
@daryanypl alrighty; I added a rescue block here.